### PR TITLE
Strip symlinks linkname only for absolute paths

### LINKED ISSUE #47: [bug] don't respect relative paths on symlinks when using strip option
While trying to decompress the LInux kernel 4.6 with `tar-fs` with the option `{strip: 1}` I've got the next error:

```
{ Error: ENOENT: no such file or directory, symlink '' -> 'tmp/linux/arch/arm/boot/dts/sun8i-a23-ippo-q8h-v1.2.dts'
    at Error (native)
  errno: -2,
  code: 'ENOENT',
  syscall: 'symlink',
  path: '',
  dest: 'tmp/linux/arch/arm/boot/dts/sun8i-a23-ippo-q8h-v1.2.dts' }
    at errorPurge (/home/piranna/Proyectos/download-manager/index.js:200:15)
    at ClientRequest.<anonymous> (/home/piranna/Proyectos/download-manager/index.js:215:11)
    at ClientRequest.g (events.js:286:16)
    at emitNone (events.js:86:13)
    at ClientRequest.emit (events.js:185:7)
    at emitAbortNT (_http_client.js:247:8)
    at _combinedTickCallback (internal/process/next_tick.js:71:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

Looking at the Linux kernet itself, that symlink [should be pointing to `sun8i-a23-q8-tablet.dts`](https://git.kernel.org/cgit/linux/kernel/git/stable/linux-stable.git/tree/arch/arm/boot/dts/sun8i-a23-ippo-q8h-v1.2.dts?id=refs/tags/v4.6.7) instead of being an empty string, and reviewing the `tar-fs` source code I think the problem is that [it's changing inconditionally all the symlinks destinations](https://github.com/mafintosh/tar-fs/blob/90b650e5b8f700e025b97364c736af3dcae588bb/index.js#L49), while it should don't do it when they are relative paths.

### PR DESCRIPTION
**Pull request by piranna**
This fixes #47

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,7 +46,12 @@ var statAll = function (fs, stat, cwd, ignore, entries) {
 var strip = function (map, level) {
   return function (header) {
     header.name = header.name.split('/').slice(level).join('/')
-    if (header.linkname) header.linkname = header.linkname.split('/').slice(level).join('/')
+
+    const linkname = header.linkname
```
Comment (mafintosh) [inline index.js]: could you switch this to a var for old node support?

**Comment by mafintosh** [comment]
very minor comment. otherwise big 👍

File: index.js (hunk 1 above)
Comment (piranna) [inline index.js]: Done, I get used lately to use `const` wherever I see it's possible :-D

**Comment by piranna** [comment]
> very minor comment. otherwise big :+1:

Thank you! :-D

File: index.js (hunk 1 above)
Comment (mafintosh) [inline index.js]: yea makes sense! i just dont wanna drop old node for existing modules :)

**Comment by mafintosh** [comment]
Released in 1.13.1 🎉

File: index.js (hunk 1 above)
Comment (piranna) [inline index.js]: > yea makes sense! i just dont wanna drop old node for existing modules :)

Makes sense, too :-)

**Comment by piranna** [comment]
> Merged #48.

Thank you! :-D It's good you accept pull-requests so fast and happily
welcome them, hopefully all devs where like you :-) You rocks, dude! :-D

**Pull request merged**